pkg/rpcclient/wsclient.go

```diff
@@ -520,6 +560,8 @@ readloop:
 				ntf.Value = new(state.AppExecResult)
 			case neorpc.NotaryRequestEventID:
 				ntf.Value = new(result.NotaryRequestEvent)
+			case neorpc.HeaderEventID:
+				ntf.Value = &block.Header{}
```

Comment:
Pay attention to the way how `case neorpc.BlockEventID:` is handled, we need exactly the same workaround for `StateRootInHeader` network setting.